include/h2o/socketpool.h

```diff
@@ -61,27 +61,25 @@ typedef struct st_h2o_socketpool_target_t {
             socklen_t len;
         } sockaddr;
     } peer;
+    void *data_for_balancer;
 
     struct {
         h2o_linklist_t sockets;
+        size_t request_count;   /* synchoronus operations should be used, counting requests on the fly */
     } _shared;
-
 } h2o_socketpool_target_t;
 
 typedef H2O_VECTOR(h2o_socketpool_target_t *) h2o_socketpool_target_vector_t;
 
-typedef size_t (*h2o_socketpool_lb_selector)(h2o_socketpool_target_vector_t *targets, void *data, int *tried);
-
-typedef void (*h2o_socketpool_lb_initializer)(h2o_socketpool_target_vector_t *targets, void **data);
-
-typedef void (*h2o_socketpool_lb_dispose_cb)(void *data);
+typedef struct st_h2o_balancer_callbacks_t h2o_balancer_callbacks_t;
 
 typedef struct st_h2o_socketpool_t {
 
     /* read-only vars */
     h2o_socketpool_target_vector_t targets;
     size_t capacity;
     uint64_t timeout; /* in milliseconds */
+    unsigned is_global : 1;
```

Comment:
`is_global_pool` of `h2o_socketpool_t` should be used instead of adding a new attribute.

I am generally not satisfied with the status of the master branch, in sense that it has if clauses checking whether it is a global pool or a host-specific pool. I hope that this PR at least does not make things more complex.

Ultimately, I think that the global socket pool should be implemented as another kind of load balancer (though I am not sure if that is possible).

In other words, you are free to (or encouraged to) refactor the existing design for simplicity.